splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java

```diff
@@ -199,12 +198,9 @@ public void reinitialize(TxnView otherTxn, boolean reinit) {
             prepared =  impl.marshallTransaction(txnView);
             activation = soi.getActivation(this, impl.getLcc());
 
-            Context statementContext = activation.getLanguageConnectionContext().getStatementContext();
+            // Push internal connection to the current context manager
             EmbedConnection internalConnection = (EmbedConnection)EngineDriver.driver().getInternalConnection();
-            ContextManager cm = internalConnection.getContextManager();
-            synchronized (cm) {
-                cm.pushContext(statementContext);
-            }
+            new EmbedConnectionContext(impl.getLcc().getContextManager(), internalConnection);
 
```

Comment:
This context is pushed to the current ContextManager created in impl.marshallTransaction(). All such contexts are popped in impl.close().